src/components/Cover/index.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import ResizeObserver from 'resize-observer-polyfill';
+import { _cs } from '@togglecorp/fujs';
+
+import Portal from '../Portal';
+
+import styles from './styles.css';
+
+interface Props {
+    className?: string;
+    children?: React.ReactNode;
+}
+
+function Cover(props: Props) {
+    const {
+        className,
+        children,
+    } = props;
+
+    const dummyRef = React.useRef<HTMLDivElement>(null);
+    const containerRef = React.useRef<HTMLDivElement>(null);
+    const resizeObserverRef = React.useRef<ResizeObserver>();
+    const mutationObserverRef = React.useRef<MutationObserver>();
+    const callbackRef = React.useRef<number | undefined>();
+
+    const queueSync = React.useCallback(() => {
+        if (callbackRef.current) {
+            window.cancelIdleCallback(callbackRef.current);
+        }
+
+        callbackRef.current = window.requestIdleCallback(() => {
```

Comment:
Looks like this doesn't work on safari at all!

Reply:
https://caniuse.com/requestidlecallback

Reply:
I've added the polyfills

Reply:
Have you imported the poly-fill and set it as side-effect? Maybe we can add this as peer-dependency?